_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,227 @@
+---
+name: iframe has a unique accessible name
+
+description: |
+ The accessible name for each iframe is unique
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to `iframe` elements that are [included in the accessibility tree](#included-in-the-accessibility-tree) and have a [non-empty](#non-empty) [accessible name](#accessible-name).
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is unique across accessible names of all applicable `iframe` within the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees), where the `src` attributes are different.
+
+**Note:** 
+- An `iframe` creates an embedded browsing context. The content of this browsing context is not part of the [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) that it is embedded in.
+
+## Assumptions
+
+There are no good use cases for repeating the [accessible name](#accessible-name) for different `iframe`.
```

Comment:
suggest edit: There are no good use cases for using the same accessible name for iframes with different sources.